authservice/src/main/java/auth/proximity/authservice/controller/AuthController.java

```diff
@@ -190,7 +198,24 @@ public ResponseEntity<UserInfoResponse> getUserInfo(@RequestParam String email)
         UserInfoResponse userInfoResponse = userService.getUserInfo(email);
         return ResponseEntity.ok(userInfoResponse);
     }
+    @PutMapping("/update/info")
+    ResponseEntity<ResponseDto> updateUserInfo(@AuthenticationPrincipal UserDetailsImpl userDetails, @RequestBody UserUpdateRequest userUpdateRequest) {
```

Comment:
This is redundant. Use role base to restrict the endpoint and in that case you wouldn't need to check if the principal is null.

If the principal is null, the authentication/authorization filter will bounce it and it won't even get to ur controller.